### ISSUE TITLE
[Bug]: Trained EfficientAd network can't detect any anomaly

### ISSUE THREAD

**Issue by westwind1027** [open-edge-platform/anomalib#2538]
### Training code
from anomalib.data import MVTec
from anomalib.engine import Engine
from anomalib.models import EfficientAd

datamodule = MVTec(
    root="./datasets/MVTec",  # Path to download/store the dataset
    category="bottle",  # MVTec category to use
    train_batch_size=1,  # Number of images per training batch
    eval_batch_size=32,  # Number of images per validation/test batch
    num_workers=8,  # Number of parallel processes for data loading
)

model = EfficientAd()

engine = Engine(max_epochs=10)  # Train for 10 epochs

engine.fit(datamodule=datamodule, model=model)

### Lightning inference code
from pathlib import Path

from anomalib.data import PredictDataset
from anomalib.engine import Engine
from anomalib.models import EfficientAd

model = EfficientAd()
engine = Engine()

dataset = PredictDataset(
    path=Path("path/to/test/images"),
    image_size=(256, 256),
)

predictions = engine.predict(
    model=model,
    dataset=dataset,
    ckpt_path="path/to/model.ckpt",
)

if predictions is not None:
    for prediction in predictions:
        image_path = prediction.image_path
        anomaly_map = prediction.anomaly_map  # Pixel-level anomaly heatmap
        pred_label = prediction.pred_label  # Image-level label (0: normal, 1: anomalous)
        pred_score = prediction.pred_score  # Image-level anomaly score

### Dataset

MVTec

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

1. install anomalib with: pip install anomalib==2.0.0b2
2. run training code as the instruction of "https://anomalib.readthedocs.io/en/latest/markdown/get_started/anomalib.html".
3. run lightning inference code as the instruction of "https://anomalib.readthedocs.io/en/latest/markdown/get_started/anomalib.html".
4. get an empty "predictions" when inference an anomaly bottle image.

### OS information

OS information:
- OS: win11 pro
- Python version: 3.11.9
- Anomalib version: 2.0.0b2
- PyTorch version: 2.3.1+cu121
- CUDA/cuDNN version:  12.6
- GPU models and configuration: 1x GeForce RTX 4080

### Expected behavior

There should be some anomaly data in "predictions".

### Pip/GitHub

pip

### What version/branch did you use?

2.0.0b2

### Configuration YAML

```yaml
default yaml configuration.
```

### Logs

```shell
No logs.
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by alexriedel1** [comment]
Whats the length of your dataset:  `len(dataset)`?

**Comment by westwind1027** [comment]
> Whats the length of your dataset: `len(dataset)`?

len(dataset) is 1.

**Comment by alexriedel1** [comment]
Sorry, but I cannot reproduce the error. If I use your training and inferencing setup I get:

```
import matplotlib.pyplot as plt

if predictions is not None:
    for prediction in predictions:
        image_path = prediction.image_path
        anomaly_map = prediction.anomaly_map # Pixel-level anomaly heatmap
        pred_label = prediction.pred_label # Image-level label (0: normal, 1: anomalous)
        pred_score = prediction.pred_score # Image-level anomaly score

        print(pred_label)
        print(pred_score)
        plt.imshow(anomaly_map.squeeze())
        plt.show()
```

```
tensor([True])
tensor([[0.5107]])
```
![Image](https://github.com/user-attachments/assets/26b12143-de9d-4e49-ac98-c09d5ab64e90)

**Comment by westwind1027** [comment]
Thank you for taking time to reproduce this issue. 
The EfficientAD works well when I downgrade Anomalib from 2.0.0b2 to 1.2.0.

![Image](https://github.com/user-attachments/assets/8c29af98-9c70-4c03-9404-040d9c2e80ff)

When I upgrade back to 2.0.0b2, it still can't work. Below are the train capture and infer capture:

![Image](https://github.com/user-attachments/assets/43ec276a-93d5-4e36-bbff-0e32cd7cb2d3)

![Image](https://github.com/user-attachments/assets/687e1173-17c3-4e6a-9933-d81df95aabd4)

**Comment by Woodthorne** [comment]
I have a similar issue, though I am using the default checkpoint path when predicting. My dataset has a length of 1, and after running engine.predict the resulting predictions is an empty list of length 0.

**Comment by AndreaBrg** [comment]
Any update on this?

I'm having the same issue and I tried to debug it a bit.
The predictions are present up untill the end of this function https://github.com/openvinotoolkit/anomalib/blob/c2a62c32204af8dc25b3d108505ac98a8dc80f0d/src/anomalib/post_processing/one_class.py#L232
which is called in https://github.com/Lightning-AI/pytorch-lightning/blob/41315bc79f1b549d793e326baccae8dad63a6c51/src/lightning/pytorch/loops/prediction_loop.py#L268 but after that line the lightning's funtion exists and does not save the predictions.

Any idea?
